### PR TITLE
Warnings for particular tokens

### LINKED ISSUE #316: Disable ability to transfer USDC and DAI from BSC
If a user tries to choose Binance-Peg USDC or Binance-Peg DAI tokens to transfer through the bridge from BSC a modal with a text like the following window must appear:

- Bridging of Binance-Peg USDC tokens to the Gnosis Chain is disabled, please use [Xpollinate](https://xpollinate.io/) bridge instead.
- Bridging of Binance-Peg DAI tokens to the Gnosis Chain is disabled, please use [Xpollinate](https://xpollinate.io/) bridge instead.

Address of USDC on BSC: https://bscscan.com/token/0x8ac76a51cc950d9822d68b83fe1ad97b32cd580d
Address of DAI on BSC: https://bscscan.com/token/0x1af3f329e8be154074d8769d1ffa4ee058b1dbc3

Since the Connext protocol now has enough liquidity for such transfers, these changes must simplify the UX.

Transfers for the corresponding bridged tokens https://blockscout.com/xdai/mainnet/token/0xD10Cc63531a514BBa7789682E487Add1f15A51E2 and https://blockscout.com/xdai/mainnet/token/0xFc8B2690F66B46fEC8B3ceeb95fF4Ac35a0054BC should be still allowed.

### LINKED ISSUE #317: Disable ability to transfer set of stable tokens from GC to BSC
If a user tries to choose USDC/USDT (with Ethereum Mainnet as the origin) or WXDAI/DAI tokens to transfer through the bridge from the Gnosis Chain to BSC a modal with a text like the following window must appear:

- Bridging of USDC tokens to Binance Smart Chain is disabled, please use [Xpollinate](https://xpollinate.io/) bridge instead.
- Bridging of USDT tokens to Binance Smart Chain is disabled, please use [Xpollinate](https://xpollinate.io/) bridge instead.
- Bridging of WXDAI tokens to Binance Smart Chain is disabled, please use [Xpollinate](https://xpollinate.io/) bridge instead.
- Bridging of DAI tokens to Binance Smart Chain is disabled, bridge tokens back to the Ethereum Mainnet and use [Xpollinate](https://xpollinate.io/) to get Binance-Peg DAI.

Address of USDC on GC: https://blockscout.com/xdai/mainnet/token/0xDDAfbb505ad214D7b80b1f830fcCc89B60fb7A83
Address of WXDAI on GC: https://blockscout.com/xdai/mainnet/token/0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d
Address of DAI on GC: https://blockscout.com/xdai/mainnet/token/0x44fA8E6f47987339850636F88629646662444217
Address of USDT on GC: https://blockscout.com/xdai/mainnet/token/0x4ECaBa5870353805a9F068101A40E0f32ed605C6

Since the Connext protocol now has enough liquidity for such transfers, these changes must simplify the UX.

Transfers for the corresponding bridged tokens
- USDC, https://bscscan.com/token/0x7faf9f7b2006124599cc4f8fb556a6d46015fefe
- USDT, https://bscscan.com/token/0x0e76a94910b951298845518b3762adad81b2be78
- WXDAI, https://bscscan.com/token/0x2c764dda26e525263f666540e9cb94ee719d18c0
- DAI, https://bscscan.com/token/0xe7c778033842504c8d6dcd7d0900b8fa4a4974a6

should be still allowed.

### LINKED ISSUE #318: Restrict some tokens to be bridged through ETH-BSC bridge
In order to reduce number of users mistakes, restrict bridging of tokens initially bridged from GC on the ETH-BSC direction.

The following message must appear and the user must not be able to make the transfer:
- Bridging of tokens initially bridged from the Gnosis Chain is not allowed in this direction. Bridge them back to the Gnosis Chain and then to the Ethereum Mainnet.

In order to check if a token is bridged from the Gnosis Chain, it is necessary to call the method `nativeTokenAddress` from [the BSC-GC OmniBrdige contract](https://bscscan.com/address/0xF0b456250DC9990662a6F25808cC74A6d1131Ea9#readProxyContract) and pass the address of the token in BSC. If it returns non-zero value, the token was bridged from the Gnosis Chain.

### LINKED ISSUE #319: Extra warnings for ETH-BSC direction
If a user tries to bridge on of the following tokens USDC, USDT, DAI, BNB, WETH through ETH-BSC direction the following message must appear in a window to warn the user about possible error:

- Bridging this Binance-Peg token to the ETH mainnet DOES NOT mint an official ERC20 token.
- Bridging this token DOES NOT mint the Binance-Peg token on the BSC

### PR DESCRIPTION
**Pull request by dan13ram**
closes #316 
closes #317 
closes #318 
closes #319

Deployed at https://sokol-omnibridge.web.app/

### CODE REVIEW

File: packages/dapp/src/components/warnings/BSCGCTokenWarnings.jsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  );
+};
+
+export const isGCStableToken = token => {
+  if (!token) return false;
+  const { address, chainId } = token;
+  return chainId === 100 && GC_STABLE_ASSETS.includes(address.toLowerCase());
+};
+
+export const BSCGCTokenWarnings = ({ token, noShadow = false }) => {
```
Comment (dan13ram) [inline packages/dapp/src/components/warnings/BSCGCTokenWarnings.jsx:40]: handles warnings as per https://github.com/omni/omnibridge-ui/issues/316

File: packages/dapp/src/components/warnings/BSCETHTokenWarnings.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  return chainId === 56 && BSC_ASSETS.includes(address.toLowerCase());
+};
+
+const isETHAsset = token => {
+  if (!token) return false;
+  const { address, chainId } = token;
+  return chainId === 1 && ETH_ASSETS.includes(address.toLowerCase());
+};
+
+export const BSCETHTokenWarnings = ({ token, noShadow = false }) => {
```
Comment (dan13ram) [inline packages/dapp/src/components/warnings/BSCETHTokenWarnings.jsx:34]: handles warnings as per https://github.com/omni/omnibridge-ui/issues/319

File: packages/dapp/src/components/warnings/GCOriginOnBSCTokenWarning.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { Alert, AlertIcon, Flex, Text } from '@chakra-ui/react';
+import { useTokenGCOriginOnBSC } from 'hooks/useTokenDisabled';
+import React from 'react';
+
+export const GCOriginOnBSCTokenWarning = ({ token, noShadow = false }) => {
```
Comment (dan13ram) [inline packages/dapp/src/components/warnings/GCOriginOnBSCTokenWarning.jsx:5]: handles warnings as per #318

File: packages/dapp/src/hooks/useTokenDisabled.jsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    } finally {
+      setFetching(false);
+    }
+  }, [token, bridgeDirection]);
+
+  useEffect(() => load(), [load]);
+  return { fetching, isToken };
+};
+
+export const useTokenDisabled = token => {
```
Comment (dan13ram) [inline packages/dapp/src/hooks/useTokenDisabled.jsx:55]: this hook is used to check if the token bridging is disabled for various reasons.

**Comment by akolotov** [comment]
Is it possible to avoid doubled message for STAKE token?

image

**Comment by dan13ram** [comment]
Thanks @akolotov. This should be fixed now.
Screenshot from 2022-02-11 18:45:13

**Review by akolotov — CHANGES_REQUESTED** [review]
Consider to change the warning statement

**Pull request merged**